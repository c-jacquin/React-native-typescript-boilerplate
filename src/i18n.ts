import 'intl'
import { addLocaleData } from 'react-intl'

import enLocaleData from 'react-intl/locale-data/en'
import frLocaleData from 'react-intl/locale-data/fr'

import enTranslationMessages from '../i18n/en.json'
import frTranslationMessages from '../i18n/fr.json'

import config from 'config'

/**
 * add helpers to format date strings and all the i18n magic
 */
addLocaleData(enLocaleData)
addLocaleData(frLocaleData)

export interface TranslationMessages {
    en: any
    fr: any
}

export const formatTranslationMessages = (
    locale: string,
    messages: any
): TranslationMessages => {
    const defaultFormattedMessages: any =
        locale !== config.LANGUAGE.DEFAULT_LOCALE
            ? formatTranslationMessages(
                  config.LANGUAGE.DEFAULT_LOCALE,
                  enTranslationMessages
              )
            : {}
    const formattedMessages: TranslationMessages | any = {}
    const messageKeys = Object.keys(messages)

    for (const messageKey of messageKeys) {
        formattedMessages[messageKey] =
            locale === config.LANGUAGE.DEFAULT_LOCALE
                ? messages[messageKey]
                : (formattedMessages[messageKey] =
                      messages[messageKey] ||
                      defaultFormattedMessages[messageKey])
    }

    return formattedMessages
}

export const translationMessages: TranslationMessages = {
    en: formatTranslationMessages('en', enTranslationMessages),
    fr: formatTranslationMessages('fr', frTranslationMessages),
}
