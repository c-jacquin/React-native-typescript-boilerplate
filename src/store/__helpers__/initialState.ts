import { AppState } from '../types'
import { initialState as languageState } from '../language/reducer'
// Import state here

const initialState: AppState = {
    language: languageState,
    // Insert state here
}

export default initialState
