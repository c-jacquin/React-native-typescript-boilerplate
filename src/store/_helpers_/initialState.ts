import { AppState } from '../types'
import { initialState as languageState } from '../language/reducer'
import { initialState as navigationState } from '../navigation/reducer'
// Import state here

const initialState: AppState = {
    language: languageState,
    navigation: navigationState,
    // Insert state here
}

export default initialState
