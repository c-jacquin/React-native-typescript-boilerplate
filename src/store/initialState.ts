import { AppState } from './types'
import { initialState as languageState } from './language/reducer'
import { initialState as navigationState } from './navigation/reducer'
import { initialState as pushNotificationState } from './pushNotification/reducer'
// Import state here

const initialState: AppState = {
    language: languageState,
    navigation: navigationState,
    pushNotification: pushNotificationState,
    // Insert state here
}

export default initialState
