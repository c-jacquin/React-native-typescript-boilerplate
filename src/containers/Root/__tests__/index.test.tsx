import React from 'react'
import { Provider, Store } from 'react-redux'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import { Root, mapStateToProps } from '../index'

import configureStore from 'redux-mock-store'
import { translationMessages } from 'i18n'
import initialState from 'store/__helpers__/initialState'

const mockStore = configureStore()

describe('Root component', () => {
    let store = mockStore({})
    const mockProps = {
        getLocale: () => ({ type: 'TEST' }),
        nav: {
            index: 0,
            routes: [
                {
                    key: 'test',
                    routeName: 'Home',
                },
            ],
        },
    }
    const spy = jest.spyOn(mockProps, 'getLocale')

    const tree = renderer.create(
        <Provider store={store}>
            <IntlProvider locale={'en'} messages={translationMessages.en}>
                <Root {...mockProps} />
            </IntlProvider>
        </Provider>
    )
    beforeEach(() => {
        store = mockStore({})
    })

    it('should renders correctly', () => {
        expect(tree).toBeDefined()
    })

    it('should call getLocale prop', () => {
        expect(spy).toHaveBeenCalled()
    })

    describe('mapStateToPropsFunction', () => {
        it('should return the navigation state in an object', () => {
            expect(mapStateToProps(initialState, null)).toEqual({
                nav: initialState.navigation,
            })
        })
    })
})
