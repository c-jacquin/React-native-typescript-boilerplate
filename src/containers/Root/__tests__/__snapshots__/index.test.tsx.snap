// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component should renders correctly 1`] = `<View />`;

exports[`Root component should renders correctly without getLocale prop 1`] = `<View />`;
